[CI] AnalyzerTests testRerankFieldsInvalidTypes failing

**Build Scans:**
- [elasticsearch-periodic #9797 / release-tests](https://gradle-enterprise.elastic.co/s/n3n2jjgpn2vlk)
- [elasticsearch-periodic #9790 / release-tests](https://gradle-enterprise.elastic.co/s/rjlm3tws4djwg)
- [elasticsearch-periodic #9783 / release-tests](https://gradle-enterprise.elastic.co/s/43yudvrgrylqs)
- [elasticsearch-periodic #9776 / release-tests](https://gradle-enterprise.elastic.co/s/sbv5ahvi6vwt6)
- [elasticsearch-periodic #9769 / release-tests](https://gradle-enterprise.elastic.co/s/ekdrqupibxzlc)
- [elasticsearch-periodic #9762 / release-tests](https://gradle-enterprise.elastic.co/s/lxv4rpy74nzf4)
- [elasticsearch-periodic #9755 / release-tests](https://gradle-enterprise.elastic.co/s/bw3i4vgtyx6ke)
- [elasticsearch-periodic #9748 / release-tests](https://gradle-enterprise.elastic.co/s/catdcc4uncbt4)
- [elasticsearch-periodic #9741 / release-tests](https://gradle-enterprise.elastic.co/s/xo4jhecy3rj62)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.analysis.AnalyzerTests.testRerankFieldsInvalidTypes" -Dtests.seed=AFD61C36C570FA67 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=ar-SD -Dtests.timezone=Jamaica -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.analysis.AnalyzerTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRerankFieldsInvalidTypes'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a string containing "rerank field must be a valid string, numeric or boolean expression, found [dense_vector]"
     but: was "Found 1 problem
line 1:68: Cannot use field [dense_vector] with unsupported type [dense_vector]"
```

**Issue Reasons:**
- [main] 9 consecutive failures in step release-tests
- [main] 9 failures in test testRerankFieldsInvalidTypes (3.5% fail rate in 256 executions)
- [main] 9 failures in step release-tests (100.0% fail rate in 9 executions)
- [main] 9 failures in pipeline elasticsearch-periodic (100.0% fail rate in 9 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/AnalyzerTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/AnalyzerTests.java
@@ -123,6 +123,7 @@ import static org.elasticsearch.xpack.esql.analysis.AnalyzerTestUtils.defaultEnr
 import static org.elasticsearch.xpack.esql.analysis.AnalyzerTestUtils.indexWithDateDateNanosUnionType;
 import static org.elasticsearch.xpack.esql.analysis.AnalyzerTestUtils.loadMapping;
 import static org.elasticsearch.xpack.esql.analysis.AnalyzerTestUtils.tsdbIndexResolution;
+import static org.elasticsearch.xpack.esql.core.plugin.EsqlCorePlugin.DENSE_VECTOR_FEATURE_FLAG;
 import static org.elasticsearch.xpack.esql.core.tree.Source.EMPTY;
 import static org.elasticsearch.xpack.esql.core.type.DataType.DATETIME;
 import static org.elasticsearch.xpack.esql.core.type.DataType.DATE_NANOS;
@@ -3755,7 +3756,9 @@ public class AnalyzerTests extends ESTestCase {
     }
 
     public void testRerankFieldsInvalidTypes() {
-        List<String> invalidFieldNames = List.of("date", "date_nanos", "ip", "version", "dense_vector");
+        List<String> invalidFieldNames = DENSE_VECTOR_FEATURE_FLAG.isEnabled()
+            ? List.of("date", "date_nanos", "ip", "version", "dense_vector")
+            : List.of("date", "date_nanos", "ip", "version");
 
         for (String fieldName : invalidFieldNames) {
             LogManager.getLogger(AnalyzerTests.class).warn("[{}]", fieldName);

